course4/homework-string/string.hh

```diff
@@ -1,10 +1,38 @@
 #pragma once
+#include <memory>
+#include <string.h>
```

Comment:
You don't use string.h and stdexcept in this file, therefore you shouldn't include them here. This will only increase the compilation time for the users of the class.